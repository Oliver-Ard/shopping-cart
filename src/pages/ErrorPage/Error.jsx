import { Link } from "react-router-dom";

function Error() {
	return (
		<>
			<h1>Nothing to see here!</h1>
			<Link to="/">Home</Link>
		</>
	);
}
export default Error;
