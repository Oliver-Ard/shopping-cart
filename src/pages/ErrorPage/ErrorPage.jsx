import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import styled from "styled-components";

const StyledError = styled.div`
	text-align: center;
	padding: 5rem 2rem;

	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;

	h1 {
		color: var(--clr-error);
		margin-bottom: 0.5rem;
	}

	p {
		font-size: var(--font-size-500);
		font-weight: 500;
		margin-bottom: 1rem;
	}

	a {
		background: var(--clr-error);
		color: var(--clr-light);
		font-weight: 500;
		padding: 0.5em 1em;
		border-radius: var(--border-radius);

		box-shadow: var(--box-shadow-2);

		transition:
			transform 150ms ease-in-out,
			box-shadow 150ms ease-in-out;

		&:hover {
			transform: scale(0.95);
			box-shadow: none;
		}

		&:active {
			opacity: 0.9;
		}
	}
`;

function ErrorPage({ path, locationName }) {
	return (
		<StyledError>
			<h1>Oops! Error: 404</h1>
			<p>We can&apos;t seem to find the page you&apos;re looking for.</p>
			<Link to={path}>{locationName}</Link>
		</StyledError>
	);
}
export default ErrorPage;

ErrorPage.propTypes = {
	path: PropTypes.string,
	locationName: PropTypes.string,
};
