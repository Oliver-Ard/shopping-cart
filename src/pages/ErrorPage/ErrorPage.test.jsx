/* global describe, it, expect, beforeEach */

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import ErrorPage from "./ErrorPage";

describe("ErrorPage component", () => {
	function renderErrorPage() {
		return render(
			<BrowserRouter>
				<ErrorPage path="/" locationName="Home" />
			</BrowserRouter>
		);
	}

	beforeEach(() => {
		renderErrorPage();
	});

	it("renders all elements", () => {
		const { container } = render(
			<BrowserRouter>
				<ErrorPage path="/" locationName="Home" />
			</BrowserRouter>
		);
		expect(container).toMatchSnapshot();
	});

	it("navigates to home when the link is clicked", async () => {
		const user = userEvent.setup();

		const linkEl = screen.getByRole("link", { name: /home/i });
		await user.click(linkEl);
		expect(window.location.pathname).toBe("/");
	});
});
