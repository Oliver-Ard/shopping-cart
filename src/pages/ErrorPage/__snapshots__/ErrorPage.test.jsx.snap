// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage component > renders all elements 1`] = `
<div>
  <div
    class="sc-aYaIB eDhQlh"
  >
    <h1>
      Oops! Error: 404
    </h1>
    <p>
      We can't seem to find the page you're looking for.
    </p>
    <a
      href="/"
    >
      Home
    </a>
  </div>
</div>
`;
